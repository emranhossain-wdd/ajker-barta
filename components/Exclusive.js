import { formatDistanceToNow } from 'date-fns';
import { useRouter } from 'next/router';
import React, { useState } from 'react'
import styles from '../styles/exclusive.module.css'
const Exclusive = ({ newses }) => {
  const latest = newses?.reverse()
  const router = useRouter()
  const [isLetest, setIsLetest] = useState(true)
  let count = 1
  return (
    <div data-testid='exclusiveId' className='pb-5'>
      <div className="container">
        <div className="grid lg:grid-cols-2 gap-3">
          <div>
            <button onClick={() => setIsLetest(true)} className={isLetest ? 'py-1 slim-border rounded-sm mb-3 px-4 border-blue-400 mr-2 bg-blue-500 text-white' : 'py-1 slim-border rounded-sm mb-3 px-4 bg-white border-blue-400 mr-2'}>Letest</button>
            <button onClick={() => setIsLetest(false)} className={!isLetest ? 'py-1 slim-border rounded-sm mb-3 px-4 border-blue-400 mr-2 bg-blue-500 text-white' : 'py-1 slim-border rounded-sm mb-3 px-4 bg-white border-blue-400 mr-2'}>Trending</button>
            {
              isLetest ? <div className='grid sm:grid-cols-2 gap-3'>
                {
                  latest?.slice(0, 8).map(news => <div onClick={() => router.push(`/${news.category}/${news.subCategory}/${news?._id}`)} className=' relative p-6 shadow-md rounded-r-lg border-r-4 border-red-600 cursor-pointer' key={news._id}>
                    <h1 className='leading-7 relative z-10 text-lg font-medium text-gray-800 hover:text-red-600 transition-colors duration-300 cursor-pointer'>{news?.heading}</h1>
                    <p style={{ transform: "translate(-50%, -50%)", }} className='absolute top-1/2 left-3/4 text-8xl font-bold text-blue-100 '>{count++}</p>
                  </div>)
                }
              </div> : <div className='grid sm:grid-cols-2 gap-3'>
                {
                  latest?.slice(8, 16).map(news => <div onClick={() => router.push(`/news/${news._id}`)} className=' relative p-6 shadow-md rounded-r-lg border-r-4 border-red-600 cursor-pointer' key={news._id}>
                    <h1 className='leading-7 relative z-10 text-lg font-medium text-gray-800 hover:text-red-600 transition-colors duration-300 cursor-pointer'>{news?.heading}</h1>
                    <p style={{ transform: "translate(-50%, -50%)", }} className='absolute top-1/2 left-3/4 text-8xl font-bold text-blue-100 '>{count++}</p>
                  </div>)
                }
              </div>
            }

          </div>
          <div>
            <h1 className={styles.exclusive}>Satireday</h1>
            <div className='grid sm:grid-cols-2 gap-3'>
              {
                latest?.slice(16, 20).map(news => <div className='cursor-pointer imageEffect' onClick={() => router.push(`/${news.category}/${news.subCategory}/${news?._id}`)} key={news?._id}>
                  <div className='overflow-hidden'>
                    <img className='w-full h-32 object-cover' src={news?.images.img1} alt="" />
                  </div>
                  <h1 className='text-lg my-1 font-medium leading-5 hover:text-red-600 transition-colors duration-300 cursor-pointer'>{news?.heading}</h1>
                  <p className='text-sm leading-4'>{news?.description[0]?.slice(0, 100)}</p>
                  <p className="px-2 mt-2 py-1 font-medium text-sm rounded-full bg-gray-100 w-fit text-blue-500">{`${formatDistanceToNow(new Date(news.publishedDate))} ago`}</p>
                </div>)
              }
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Exclusive;