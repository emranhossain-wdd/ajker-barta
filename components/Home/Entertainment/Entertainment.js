import React, { useEffect, useState } from 'react';
import styles from '../../../styles/Hero.module.css';
import ChevronRightIcon from '@mui/icons-material/ChevronRight';
import Image from 'next/image';


const Entertainment = () => {
    const [enData, setEnData] = useState([]);

    useEffect(() => {
        const url = `./newsData.JSON`
        fetch(url)
            .then(res => res.json())
            .then(data => {
                setEnData(data);
            })
    }, [])


    return (
        <div className="container overflow-hidden ">
            <h2 className='text-xl text-blue-900 font-semibold mb-3'><span className='text-gray-700'>Entertainment </span><ChevronRightIcon className='text-red-500' /></h2>


            <div className="grid grid-cols-12 lg:gap-3 gap-2">

                <div className='col-span-12 lg:col-span-6'>
                    <div className='relative w-full h-full'>
                        <img className='w-full h-full' src={enData[0]?.image} alt='' />
                        <div className='absolute bottom-4 left-4 z-10'>
                            <h2 className='text-white text-2xl cursor-pointer hover:text-red-400 transition-colors duration-300'>{enData[0]?.title}</h2>
                        </div>
                        <div className={styles.overlay}></div>
                    </div>
                </div>


                <div className='col-span-12 lg:col-span-6'>

                    <div className="col-span-12">
                        <div className='relative w-full max-h-72'>
                            <img className=' object-fill h-72 w-full' src={enData[1]?.image} alt='' />
                            <div className='absolute bottom-4 left-4 z-10'>
                                <h2 className='text-white text-1xl cursor-pointer hover:text-red-400 transition-colors duration-300'>{enData[1]?.title}</h2>
                            </div>
                            <div className={styles.overlay}></div>
                        </div>
                    </div>


                    <div className='grid grid-cols-2 gap-3 mt-3'>
                        <div className="">
                            <div className='relative w-full max-h-72'>
                                <img className='object-fill h-72 w-full' src={enData[2]?.image} alt='' />
                                <div className='absolute bottom-4 left-4 z-10'>
                                    <h3 className='text-white text-1xl cursor-pointer hover:text-red-400 transition-colors duration-300'>{enData[2]?.title}</h3>
                                </div>
                                <div className={styles.overlay}></div>
                            </div>
                        </div>


                        <div className="">
                            <div className='relative w-full max-h-72'>
                                <img className='object-fill h-72 w-full' src={enData[3]?.image} alt='' />
                                <div className='absolute bottom-4 left-4 z-10'>
                                    <h3 className='text-white text-1xl  cursor-pointer hover:text-red-400 transition-colors duration-300'>{enData[3]?.title}</h3>
                                </div>
                                <div className={styles.overlay}></div>
                            </div>
                        </div>
                    </div>

                </div>

            </div>
        </div>
    );
};

export default Entertainment;
