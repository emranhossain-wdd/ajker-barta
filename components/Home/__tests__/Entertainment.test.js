import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Entertainment from '../Entertainment/Entertainment'
import { render } from '@testing-library/react'
import { testArr } from '../../__tests__/Exclusive.test'

it('render Entertainment component without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Entertainment />, div)
})

it('testing Entertainment with props', () => {
  const { getByTestId } = render(<Entertainment testArr={testArr} />)
  expect(getByTestId('entertainmentId')).toBeTruthy()
})
it('maching snapshot', () => {
  const tree = renderer.create(<Entertainment />)
  expect(tree).toMatchSnapshot()
})