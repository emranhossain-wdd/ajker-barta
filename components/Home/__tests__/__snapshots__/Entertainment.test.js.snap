// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maching snapshot 1`] = `
<div
  className="pb-5"
  data-testid="exclusiveId"
>
  <div
    className="container"
  >
    <div
      className="grid lg:grid-cols-2 gap-3"
    >
      <div>
        <button
          className="py-1 slim-border rounded-sm mb-3 px-4 border-blue-400 mr-2 bg-blue-500 text-white"
          onClick={[Function]}
        >
          Letest
        </button>
        <button
          className="py-1 slim-border rounded-sm mb-3 px-4 bg-white border-blue-400 mr-2"
          onClick={[Function]}
        >
          Trending
        </button>
        <div
          className="grid sm:grid-cols-2 gap-3"
        />
      </div>
      <div>
        <h1
          className="exclusive"
        >
          Satireday
        </h1>
        <div
          className="grid sm:grid-cols-2 gap-3"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`maching snapshot 2`] = `
<div
  className="container overflow-hidden "
  data-testid="entertainmentId"
>
  <h2
    className="text-xl text-blue-900 font-semibold mb-3"
  >
    <span
      className="text-gray-700"
    >
      Entertainment 
    </span>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium text-red-500 css-i4bv87-MuiSvgIcon-root"
      data-testid="ChevronRightIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 6 8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
      />
    </svg>
  </h2>
  <div
    className="galleryRow"
  />
</div>
`;
